common/validators.js

```diff
@@ -13,6 +13,7 @@
         return {
             require: 'ngModel',
             link: function(scope, elm, attrs, ctrl) {
+                // TODO: Wrap in catcher fn
```

Comment:
I don't think this is necessary. As well as all proposed changes to the validators for error handling.